### PR TITLE
Moose audit

### PR DESCRIPTION
**Pull request by Karol1223**
#### Summary
None

#### Purpose of change
I won't even lie to you it started because I read the tusked moose's description and it was a mess

#### Describe the solution
**Regular moose**
- Health significantly increased
- Armor pretty significantly increased
- Gestation time increased
- Calf growth time increased
- Calf description slightly edited
- Added a ``PLAYER_CLOSE`` anger trigger because you are NOT getting close to a wild moose without getting mauled or at the very least the thing running off
- Removed the ``upgrades`` field - the general idea is not 100% of the population mutates, and if we make monsters upgrade in real time we can't show that in any way. We don't do this for any other animals either to my knowledge (except maybe frogs, but if we do that's something to look into)

**Tusked moose**
- Significantly (and I mean it) increased health
- Cut armor significantly increased, very minor increase to ballistic armor
- Increased the weight and volume
- Edited the description
- Increased the gestation timer
- Calf growth time increased
- Calf description slightly edited
- Slightly lowered the speed

**Weakpoints & Bullshit**
- Added a new chest weakpoint to the quadruped weakpoint set - easier to hit for ranged weapons and provides crit modifiers for that, but increases damage against bash hits because ribs hard
- Slightly lowered the difficulty for ranged weapons to hit the head

#### Describe alternatives you've considered
- Not lowering the speed for the tusked moose

#### Additional context
The moose shall no longer be only 50% more tanky than a default zombie. Seriously what kind of bullshit was that

### CODE REVIEW

**Comment by ADekema** [comment]
How will this affect bullet damage. Bullet damage seems to be losely determined by ¨You can shoot this creature dead with one shoot during hunting¨. You might be a lot of discussion on bullet damage and ranged crits later if you just increase HP a lot. Also 500 HP for the tusked moose seems a lot for a mutant that is still only the size of a horse. That is more HP than hulks who are mend to be hyper evolved biological tanks.

**Comment by Karol1223** [comment]
> Also 500 HP for the tusked moose seems a lot for a mutant that is still only the size of a horse.

Yeah our horses haev even more absurdly low health at only 90. Most our monsters do. Hulks included, given their description (but their size is also very weird in the game values)

Something being weird doesn't mean we should accept things being weird. The tusked moose should be a powerhouse. A normal moose already is irl.

**Comment by ADekema** [comment]
> If you check the tests failing here you'll see that the game is angry at me for that (with archery, not bullets) - because you can no longer reasonably hunt moose. This should be solvable with a weakpoint set, but I gotta figure out how to add one.
This would definetly be a good idea. Otherwise when increasing the HP pool of many creatures to more reasonable levels you will end up with the whole archery damage crit debate again. I would however lower the tusk moose HP to 400 otherwise it can somehow surive a critical from a .50 BMG M2 AP which no living creature the size of a horse should be able to do.

**Comment by Karol1223** [comment]
The tests on this are giving me a headache. Someone wake me up from my slumber when archery tests are less annoying.

File: data/json/monster_weakpoints/quadruped_weakpoints_animal.json [hunk 1]
```diff
@@ -52,6 +53,21 @@
...
+        "damage_mult": { "stab": 1.25, "bullet": 1.33 },
+        "coverage": 18
+      },
+      {
+        "name": "the lungs",
+        "armor_mult": { "physical": 0.45 },
+        "difficulty": { "ranged": 3 },
+        "damage_mult": { "physical": 1.75 },
+        "coverage_mult": { "melee": 0 },
+        "coverage": 7
```
Comment (Maleclypse) [inline data/json/monster_weakpoints/quadruped_weakpoints_animal.json:69]:
```suggestion
        "coverage": 7,
        "effects": [
          {
            "effect": "bleed",
            "intensity": [ 10, 15 ],
            "duration": [ 10, 20 ],
            "chance": 75,
            "message": "The %s is shot in the lungs!",
            "damage_required": [ 1, 10 ]
          },
          {
            "effect": "bleed",
            "intensity": [ 20, 30 ],
            "duration": [ 20, 30 ],
            "chance": 85,
            "message": "The %s is shot in the lungs!",
            "damage_required": [ 11, 50 ]
          },
          {
            "effect": "bleed",
            "intensity": [ 30, 40 ],
            "duration": [ 30, 40 ],
            "chance": 95,
            "message": "The %s is shot in the lungs!",
                        "damage_required": [ 51, 100 ]
          }
        ]
```

**Comment by Maleclypse** [comment]
I'd test it but I think that level of bleed effect should allow for the moose to be removed from the damage test since that test won't gauge if it bleeds out.

**Comment by Maleclypse** [comment]
Anyway I think we can remove crits and just add multiple weakpoints to living animals with significant bleed effects.

**Comment by Karol1223** [comment]
> I'd test it in game but I think that level of bleed effect should allow for the moose to be removed from the damage test since that test won't gauge if it bleeds out.

I am leaving your changes in because I agree that bleed is a good effect to apply, but in general I was leaving this PR alone because Kevin was supposed to do something about how crits/weakpoints interact to get the tests to be passable here.

**Comment by Night-Pryanik** [comment]
@Karol1223 care to fix the tests?

**Comment by Karol1223** [comment]
> @Karol1223 care to fix the tests?

I don't know how.

Fundamentally, it's not because the moose are actually too hard to kill, it's something about archery tests not fully taking weakpoints into account. Kevin has a better idea of the actual cause, all I know is the tests aren't functioning properly and that it's beyond my skillset.

**Comment by Night-Pryanik** [comment]
Maybe just (temporarily) replace moose in tests with some other, more "killable" animal?

**Comment by Karol1223** [comment]
> Maybe just (temporarily) replace moose in tests with some other, more "killable" animal?

I'm not sure how acceptable that is. I briefly considered making a test moose that'd just be a clone of the old one used solely for tests, but that feels wrong. Tests that are supposed to test ingame values should test those values.

But at the same time, those tests shouldn't be a blocker to changing those values either. This is a shitty situation all around. I'll likely have to switch the test monster like you said, though.

**Comment by GuardianDll** [comment]
I poked a bit into changes, and into the code, and my only real suggestion is to remove/comment out the moose archery test
In detail, the previous idea of archery was "bow hunting is possible, you can kill a moose with an arrow", which were based on two things: archery having ludicrious damage on crit, and moose having relatively small amount of hp; 
At this point there is an idea flying around, that since weakpoints are a thing, we can redo the archery to be more genuine, from "you can oneshot moose with arrow with a lucky strike" to "you can hit a moose in vital organ, and it will slowly bleed to death" (we do not consider brain death just yet, but that sound reasonable)
Issues are next:
Firstly, the test is not suited for anything but instakill checks, at all, whereas here we should check the moose status some amount of time later
Secondly, our bleed effect is simply not strong enough, i locally shot a moose and waited it to bleed, and while it did deal a substantial damage (40 damage for shot, and around 200 damage for bleed, give or take), it was not enough to kill a moose; i blame bleeding effect not being dangerous enough (in 10 minutes, moose bleeding simply stopped)
Thirdly, and probably most important in context of this PR, there is no simple way to ask the game "hit the target precisely into this weakpoint". Karol says, that Kevin mentioned tests are not hooked to weakpoint stuff at all, but it's hard for me to confirm it, `deal_projectile_attack` seems to lead right where weakpoint stuff happens; It is also hard for me to tell how hard it would be to pass weakpoint id as argument for deal_projectile_attack(), i am not familiar with weakpoint code

So it happened to be the change right on the edge of the new way to handle archery, and ideally we need a separate archery test for it, something Karol cannot deliver. Since the change itself is pretty desired, i would suggest to comment mooses from the test away, if @kevingranade approve it

**Comment by kevingranade** [comment]
With a complex as this is going to be to fix I can ok just disabling the moose test for the time being and making them super hard to kill.

The right way, according to me at least,  is to make it where you're implicitly aiming at the moose's thorax where it will guarantee lung and/or heart his with a "good enough" aim and set aim to be "good enough" in the test.  In practice weak points are mostly independent of "how good your aim was" and there's no code path to tell them that a particular attack should be a really good hit.

Also as guardian points out the impact of hitting a good weak point is also too low right now and the target will not bleed out ever, much less reliably and quickly.

**Comment by Karol1223** [comment]
Right. In this case I'm just gonna need someone to guide me as to how to disable the test without straight up ripping it out, because all I can do is delete.

**Comment by Karol1223** [comment]
So uh... Is it now the same thing but from gun tests?

**Comment by Night-Pryanik** [comment]
AFAIK it's unrelated. I kicked the tests, let's see what they show.

**Comment by Karol1223** [comment]
> AFAIK it's unrelated. I kicked the tests, let's see what they show.

You kicked the windows build, I'm talking about the general build matrix

**Comment by Night-Pryanik** [comment]
I kicked it too rn.

**Comment by Karol1223** [comment]
Yeah it's reoccuring. Now the gun tests are doing the same thing fucking hell this PR is a drag.

**Comment by GuardianDll** [comment]
mon_test_shotgun_0_bullet copies from mon_cow, which probably apply weakpoints of cow to the test monster, making it hit vital parts and deal more damage than it expects. Should be fixed if you override weakpoints <https://github.com/CleverRaven/Cataclysm-DDA/blob/7acb286beeb4b07ee27ee0f45e9a9853dffa277f/data/mods/TEST_DATA/monsters.json#L419>

**Pull request merged**